test/tests/functional/pbs_multi_sched.py

```diff
@@ -647,6 +647,268 @@ def test_multi_fairshare(self):
+        # pbsuser1 job will run and other two will be queued
+        self.server.expect(JOB, {'job_state': 'R'}, id=sc2_jid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=sc2_jid3)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=sc2_jid2)
+        # pbsuser3 job will run after pbsuser1
+        self.server.expect(JOB, {'job_state': 'R'}, id=sc2_jid3)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=sc2_jid2)
+        # pbsuser2 job will run in the end
+        self.server.expect(JOB, {'job_state': 'R'}, id=sc2_jid2)
+        self.server.expect(JOB, {'job_state': 'F'}, id=sc2_jid2, extend='x')
+        # query fairshare and check usage
+        sc2_fs_user1 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER1))
+        self.assertEquals(sc2_fs_user1.usage, 3)
+        sc2_fs_user2 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER2))
+        self.assertEquals(sc2_fs_user2.usage, 3)
+        sc2_fs_user3 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER3))
+        self.assertEquals(sc2_fs_user3.usage, 3)
+        sc2_fs_user4 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER4))
+        self.assertEquals(sc2_fs_user4.usage, 1)
+        # Check the multisched 'sc2' usage file whether it's updating or not
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'False'},
+                            id='sc2')
+        sc2_J1 = Job(TEST_USER1, attrs=sc2_attr)
+        sc2_jid1 = self.server.submit(sc2_J1)
+        sc2_J2 = Job(TEST_USER2, attrs=sc2_attr)
+        sc2_jid2 = self.server.submit(sc2_J2)
+        sc2_J4 = Job(TEST_USER4, attrs=sc2_attr)
+        sc2_jid4 = self.server.submit(sc2_J4)
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
+                            id='sc2')
+        # pbsuser4 job will run and other two will be queued
+        self.server.expect(JOB, {'job_state': 'R'}, id=sc2_jid4)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=sc2_jid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=sc2_jid2)
+        # pbsuser1 job will run after pbsuser4
+        self.server.expect(JOB, {'job_state': 'R'}, id=sc2_jid1)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=sc2_jid2)
+        # pbsuser2 job will run in the end
+        self.server.expect(JOB, {'job_state': 'R'}, id=sc2_jid2)
+        self.server.expect(JOB, {'job_state': 'F'}, id=sc2_jid2, extend='x')
+        # query fairshare and check usage
+        sc2_fs_user1 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER1))
+        self.assertEquals(sc2_fs_user1.usage, 5)
+        sc2_fs_user2 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER2))
+        self.assertEquals(sc2_fs_user2.usage, 5)
+        sc2_fs_user3 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER3))
+        self.assertEquals(sc2_fs_user3.usage, 3)
+        sc2_fs_user4 = self.scheds['sc2'].query_fairshare(name=str(TEST_USER4))
+        self.assertEquals(sc2_fs_user4.usage, 3)
+
```

Comment:
I will recommend you to write one more test case that verifies the usage value remains same before and after the restart of multisched.

Reply:
Ok, do i need to write a new test case or just add restart code into this testcase itself.

Reply:
I think adding restart code and verifying usage after the restart can be added here. No need to write separate test case.

Reply:
Thanks @varunsonkar i will add here